homeassistant/scripts/benchmark/__init__.py

```diff
@@ -192,6 +192,95 @@ def yield_events(event):
     return timer() - start
 
 
+@benchmark
+async def logbook_filtering_entity_id(hass):
+    """Run a 100k state changes through a logbook entity filter."""
+    # pylint: disable=import-outside-toplevel
+    from homeassistant.components import logbook
+
+    config = {
+        "include": {
+            "domains": [
+                "automation",
+                "script",
+                "group",
+                "media_player",
+                "custom_component",
+            ],
+            "entity_globs": [
+                "binary_sensor.*_contact",
+                "binary_sensor.*_occupancy",
+                "binary_sensor.*_detected",
+                "binary_sensor.*_active",
+                "input_*",
+                "device_tracker.*_phone",
+                "switch.*_light",
+                "binary_sensor.*_charging",
+                "binary_sensor.*_lock",
+                "binary_sensor.*_connected",
+            ],
+            "entities": [
+                "test.entity_1",
+                "test.entity_2",
+                "binary_sensor.garage_door_open",
+                "test.entity_3",
+                "test.entity_4",
+            ],
+        },
+        "exclude": {
+            "domains": ["input_number"],
+            "entity_globs": ["media_player.google_*", "group.all_*"],
+        },
+    }
+
+    entity_ids = [
+        "automation.home_arrival",
+        "script.shut_off_house",
+        "binary_sensor.garage_door_open",
+        "binary_sensor.front_door_lock",
+        "binary_sensor.kitchen_motion_sensor_occupancy",
+        "switch.desk_lamp",
+        "light.dining_room",
+        "input_boolean.guest_staying_over",
+        "person.eleanor_fant",
+        "alert.issue_at_home",
+        "calendar.eleanor_fant_s_calendar",
+        "sun.sun",
+    ]
+    now = dt_util.utcnow()
+    events = [
+        _create_state_changed_event_from_old_new(
+            entity_id,
+            now,
+            {"entity_id": entity_id, "state": "old"},
+            {
+                "entity_id": entity_id,
+                "state": "new",
+                "last_changed": 1,
+                "last_updated": 1,
+            },
+        )
+        for entity_id in entity_ids
+    ]
+
+    entity_attr_cache = logbook.EntityAttributeCache(hass)
+
+    def yield_events(events):
+        # pylint: disable=protected-access
+        entities_filter = logbook._generate_filter_from_config(config)
+        size = len(events)
+        for i in range(10 ** 5):
+            event = events[i % size]
+            if logbook._keep_event(hass, event, entities_filter, entity_attr_cache):
+                yield event
+
+    start = timer()
+
+    list(logbook.humanify(hass, yield_events(events), entity_attr_cache))
```

Comment:
We should probably only benchmark running the entity ids though the filter here instead of involving logbook since any logbook change is going to make this benchmark impossible to compare to previous versions.

Reply:
Made this adjustment. 

Note that I also needed to make a small tweak to the logbook benchmarks in light of these recent changes, it was also trying to setup logbook with just `{}` which now breaks. I changed it to `logbook.CONFIG_SCHEMA({})`. 

Since I had to tweak it I also moved generation of the filter out of the `yield_events` function. Since this is really about testing the performance of `_keep_events` I figured time spent generating the entity filter shouldn't be included, even if it is only once per test.